src/partition/repart.c

```diff
@@ -1480,6 +1573,27 @@ static int context_read_definitions(
                 context->n_partitions++;
         }
 
+        /* Check that each configured verity hash/data partition has a matching verity data/hash partition. */
+
+        LIST_FOREACH(partitions, p, context->partitions) {
+                Partition *q, *s;
+
+                if (p->verity == VERITY_OFF)
+                        continue;
+
+                q = find_verity_sibling(context->partitions, p);
+                if (!q)
+                        return log_syntax(NULL, LOG_ERR, p->definition_path, 1, SYNTHETIC_ERRNO(EINVAL),
+                                          "Missing sibling partition for verity partition with VerityName=%s",
+                                          p->verity_name);
+
+                s = find_verity_sibling(q->partitions_next, p);
+                if (s)
+                        return log_syntax(NULL, LOG_ERR, p->definition_path, 1, SYNTHETIC_ERRNO(EINVAL),
+                                          "Multiple sibling partitions found for verity partition with VerityName=%s",
+                                          p->verity_name);
```

Comment:
There's never going to be thousands of partitions, so the search will always be instant, so I don't think we need to cache the result.

Reply:
Even so, still I'd like to cache it, then the code should become much simpler, I guess.

Reply:
How would the code become simpler? If I understand correctly, it would only allow us to replace subsequent calls to find_verity_sibling() with the variable instead, that doesn't seem like a major simplification.